src/rateEngine/billingDeterminants/__tests__/Demand/DailyDemandAveraged.test.ts

```diff
@@ -0,0 +1,144 @@
+import LoadProfile from '../../../LoadProfile';
+import times from 'lodash/times';
+import Demand from '../../Demand';
+
+const getLoadProfileOfOnes = () => times(8760, () => 1);
+
+const zerosByMonth = times(12, () => 0);
+
+describe('DailyDemandAveraged', () => {
+  let loadProfile: LoadProfile;
+
+  beforeEach(() => {
+    loadProfile = new LoadProfile(getLoadProfileOfOnes(), { year: 2019 });
+  });
+
+  describe('calculate', () => {
+    it('calculates demand per day averaged on a simple load profile with nothing filtered', () => {
+      const result = new Demand(
+        {
+          demandPeriod: 'daily',
+          months: [],
+          daysOfWeek: [],
+          hourStarts: [],
+          onlyOnDays: [],
```

Comment:
are these optional? does the test pass if they aren't given empty values?

Reply:
They are optional, and the tests do pass if they're omitted

Reply:
nice. i assumed that but wanted to check, spreading different bags of options into methods using TypeScript isn't something I have developed good intuitions around yet.